cypress/integration/settings.spec.js

```diff
@@ -0,0 +1,221 @@
+        cy.get('#favicon').should('have.value', IMAGE_DIR + TEST_LOGO_IMAGES[1])
+        cy.get('#shareicon').should('have.value', IMAGE_DIR + TEST_LOGO_IMAGES[2])
+    })
+
+
+    it('Should change analytics codes and have change reflect correctly on save', () => {
+        cy.get('#facebook_pixel').type(TEST_FACEBOOK_PIXEL_ID)
+        cy.get('#google_analytics').type(TEST_GOOGLE_ANALYTICS_ID)
+
+        cy.saveSettings()
+
+        cy.get('#facebook_pixel').should('have.value', TEST_FACEBOOK_PIXEL_ID)
+        cy.get('#google_analytics').should('have.value', TEST_GOOGLE_ANALYTICS_ID)
+    })
+
+    // [r, g, b] -> #RRGGBB
+    function rgbToHex(r, g, b) {
+        return "#" + ((1 << 24) + (r << 16) + (g << 8) + b).toString(16).slice(1);
+    }
+    it('Should change Primary and Secondary colors and have colors reflected on page previews', () => {
+        // Enter RGB values for primary and secondary colors
+        cy.contains('p', 'Primary').siblings('div').click()
+        cy.contains(/^r/).siblings().clear().type(TEST_PRIMARY_COLOR[0].toString())
+        cy.contains(/^g/).siblings().clear().type(TEST_PRIMARY_COLOR[1].toString())
+        cy.contains(/^b/).siblings().clear().type(TEST_PRIMARY_COLOR[2].toString())
+        cy.contains('button', 'Select').click()
+
+        cy.contains('p', 'Secondary').siblings('div').click()
+        cy.contains(/^r/).siblings().clear().type(TEST_SECONDARY_COLOR[0].toString())
+        cy.contains(/^g/).siblings().clear().type(TEST_SECONDARY_COLOR[1].toString())
+        cy.contains(/^b/).siblings().clear().type(TEST_SECONDARY_COLOR[2].toString())
+        cy.contains('button', 'Select').click()
+
+        cy.saveSettings()
+
+        // Check if selected colors are reflected upon save
+        const rgb_primary = `rgb(${TEST_PRIMARY_COLOR.join(', ')})`
+        const hex_primary = rgbToHex(...TEST_PRIMARY_COLOR)
+        const rgb_secondary = `rgb(${TEST_SECONDARY_COLOR.join(', ')})`
+        const hex_secondary = rgbToHex(...TEST_SECONDARY_COLOR)
+        cy.contains('p', 'Primary').siblings('input')
+            .should('have.value', hex_primary)
+        cy.contains('p', 'Primary').siblings('div')
+            .should('have.attr', 'style', `background: ${rgb_primary};`)
+        cy.contains('p', 'Secondary').siblings('input')
+            .should('have.value', hex_secondary)
+        cy.contains('p', 'Secondary').siblings('div')
+            .should('have.attr', 'style', `background: ${rgb_secondary};`)
+
+        // Check if page previews reflect color change
+        cy.visitLoadSettings(`/sites/${TEST_REPO_NAME}/${SAMPLE_PAGE}`)
+        cy.get('section.bp-section-pagetitle') // Page title banner
+            .should('have.css', 'background-color', rgb_primary)
+
+        // Check if home page reflects color change
+        cy.visit(`/sites/${TEST_REPO_NAME}/workspace`) // Somehow colors won't load on homepage if visiting directly
```

Comment:
Retrieval of colors happens through `useEffect`, so you might need to wait for certain cues (like checking if the custom stylesheet was created)